libraries/eosiolib/contracts/eosio/key_value.hpp

```diff
@@ -86,6 +86,20 @@
       }                                                                                                                \
    };
 
+/**
+ * @brief Macro to define a table.
+ * @details In the case where the autogenerated index names created by DEFINE_TABLE are not enough, a user can instead
+ * manually define the table and indices. This macro allows users to conveniently define an index without having to specify
+ * the index template type, as those can be large/unwieldy to type out.
+ *
+ * @param variable_name - The variable name in code of the index.
+ * @param index_name    - The index name.
+ * @param value_class   - The class the table stores.
+ * @parma member_name   - The name of the member pointer used for the index.
+ */
+#define NAMED_INDEX(variable_name, index_name, value_class, member_name)                                               \
```

Comment:
You can avoid requiring value_class by creating a value_type typedef in kv_table or some such.